packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -360,40 +355,15 @@ async function _getDashboardStats(app) {
     console.error(error)
   }
 
-  dashboard.alarms = alarms.reduce((acc, { $object, body, time }) => {
-    try {
-      const [, value, name] = body.match(ALARM_BODY_REGEX)
-
-      let object
-      try {
-        object = app.getObject($object)
-      } catch (error) {
-        console.error(error)
-        object = {
-          type: 'unknown',
-          uuid: $object,
-        }
-      }
-
-      acc.push({
-        name,
-        object: {
-          type: object.type,
-          uuid: object.uuid,
-        },
-        timestamp: time,
-        value: +value,
-      })
-    } catch (error) {
-      console.error(error)
-    }
-
-    return acc
-  }, [])
   return dashboard
 }
 const getDashboardStats = throttle(_getDashboardStats, 6e4, { trailing: false, leading: true })
 
+const messagesFilter =
+  ({ onlyAlarms = false, excludeAlarms = false } = {}) =>
+  _ =>
+    _.type === 'message' && (onlyAlarms ? _.name === 'ALARM' : excludeAlarms ? _.name !== 'ALARM' : true)
+
```

Comment:
I am not a fan of using unnamed variable. What about destructuring ? 

also the imbricated ternary operator is quite long. Since messageFilter is either used with onlyalarm === true and excludeAlarms===false, or used with onlyalarm === false and excludeAlarms===true , I think we could split it in 2 filters, for example :  `keepAlarmMessages` and `keepNonAlarmMessages`

```suggestion
const  keepAlarmMessages = ({type,name})=>  type === 'message' && name === 'ALARM' 
const  keepNonAlarmMessages = ({type,name}) => type === 'message' && name !== 'ALARM' 
```